tests/exceptions/exceptions.h

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Common code for exceptions tests
+//
+*******************************************************************************/
+
+#ifndef __SYCL_CTS_TEST_EXCEPTIONS_H
+#define __SYCL_CTS_TEST_EXCEPTIONS_H
+
+#include "../common/common.h"
+#include <vector>
+
+namespace exceptions {
```

Comment:
Could you clarify the purpose of `namespace exceptions`, please?

Reply:
I've add clarifications for this `namespace` in this commit: 169e0b181a4e405853c0139fe67fc1fed264e596

Reply:
The comment doesn't explain why this namespace is needed. Can we remove it?

Reply:
We using this namespace to avoid some redefinition errors or situations where we have variable/function with the same name

Reply:
@bader Let's assume we doesn't provide any namespace for `tests/foo/foo.h` and `tests/boo/boo.h`. Let's then assume we have same variable/function name in both headers. Wouldn't it break ODR-rule for a `test_all` executable, sometimes without even compilation failure?

Reply:
The variable is declared as `static`, so it doesn't break ODR-rule.
As this header is included to a single test, I suggest we move the `all_err_codes` definition to tests/exceptions/exceptions_make_error_code.cpp and do not add tests/exceptions/exceptions.h file.

Reply:
We are going to use it in tests that we will provide soon. Should we remove header now and add it again later?

Reply:
> We using this namespace to avoid some redefinition errors or situations where we have variable/function with the same name

> We are going to use it in tests that we will provide soon. 

For this use case, I think something like this would work better:

```suggestion
namespace {

constexpr std::array<sycl::errc, 15> all_err_codes{
```

Reply:
if we will use `unnamed namespace` with `all_err_codes` variable within a header file. ODR-using from multiple translation units (from future tests for exceptions) violates the one-definition rule because the definition of `all_err_codes` is not identical in all translation units.

Reply:
You can add `inline` to `all_err_codes` definition if future tests add ODR-uses.

I still don't see any value in using namespaces for `all_err_codes` definition, but it's your preference, please, avoid using generic name like exception as it doesn't solve "redefinition errors or situations where we have variable/function with the same name".
I suggest we either don't use namespace at all or use existing approaches: `sycl_cts` or anonymous namespace.

Reply:
Changed to anonymous namespace.